homeassistant/components/season/sensor.py

```diff
@@ -53,34 +109,110 @@
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Display the current season."""
     if None in (hass.config.latitude, hass.config.longitude):
         _LOGGER.error("Latitude or longitude not set in Home Assistant config")
         return False
 
+    time_zone = hass.config.time_zone
     latitude = util.convert(hass.config.latitude, float)
     _type = config.get(CONF_TYPE)
     name = config.get(CONF_NAME)
 
+    if time_zone is None:
+        _LOGGER.warning(
+            "Time zone not set in Home Assistant configuration, UTC will be returned"
+        )
+
     if latitude < 0:
         hemisphere = SOUTHERN
     elif latitude > 0:
         hemisphere = NORTHERN
     else:
         hemisphere = EQUATOR
 
+    if EQUATOR in hemisphere:
+        _LOGGER.warning(
+            "Season cannot be determined for equator, 'unknown' state will be shown"
+        )
+
     _LOGGER.debug(_type)
-    add_entities([Season(hass, hemisphere, _type, name)], True)
 
-    return True
+    season_data = SeasonData(hemisphere, _type, time_zone)
+
+    await season_data.async_update()
+
+    entities = []
+    for description in SENSOR_TYPES:
+        if description.key in ENTITY_SEASON:
+            entities.append(Season(season_data, description, name))
+        elif hemisphere not in EQUATOR:
+            entities.append(Season(season_data, description, name))
```

Comment:
Kind of nit-picky, but...

These conditionals read oddly to me. At a glance, I would think they are list-based comparisons instead of string comparisons. Changing from `in` to `==` would help readability.

There are other places where this could be applied, too.

Reply:
I applied your suggestion along with conflicts with current dev version... Thank you :)